### PR TITLE
add change log for TiKV.

### PR DESCRIPTION
**Pull request by hicqu**
Add `HISTORY.md` so that we can check history changes easily.

### CODE REVIEW

**Comment by hicqu** [comment]
@siddontang @disksing Do you have any suggestions for the format or similar things?

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# TiKV Change Log
```
Comment (BusyJay) [inline CHANGELOG.md]: Please add a link to the change logs for PD/TiDB/TiKV.

**Comment by Hoverbear** [comment]
@hicqu Do you think it might be useful to link to the PRs/Issues that were closed by points in the log?

**Comment by Hoverbear** [comment]
I found this attempt to make a standard format: https://keepachangelog.com/en/1.0.0/

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+All notable changes to this project will be documented in this file.
+See also [TiDB change log][tidb_change_log] and [PD change log][pd_change_log].
+
+[tidb_change_log]: https://github.com/pingcap/tidb/CHANGELOG.md
+[pd_change_log]: https://github.com/pingcap/pd/CHANGELOG.md
```
Comment (lilin90) [inline CHANGELOG.md]: The two links do not work. Please update.

File: CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,132 @@
+# TiKV Change Log
+All notable changes to this project will be documented in this file.
```
Comment (lilin90) [inline CHANGELOG.md]: will be -> are?

File: CHANGELOG.md [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+See also [TiDB change log][tidb_change_log] and [PD change log][pd_change_log].
+
+[tidb_change_log]: https://github.com/pingcap/tidb/CHANGELOG.md
+[pd_change_log]: https://github.com/pingcap/pd/CHANGELOG.md
+
+## [2.0.0-GA] - 2018-04-27
+### New Features
+* `tikv-ctl` now can recover mvcc data corruption.
+* Support column families for raw key/value API.
+### Improves
```
Comment (lilin90) [inline CHANGELOG.md]: Improves -> Improvements

Please modify all "Improves" in the following text.

File: CHANGELOG.md [hunk 5]
```diff
@@ -0,0 +1,132 @@
+# TiKV Change Log
+All notable changes to this project will be documented in this file.
+See also [TiDB change log][tidb_change_log] and [PD change log][pd_change_log].
+
+[tidb_change_log]: https://github.com/pingcap/tidb/CHANGELOG.md
+[pd_change_log]: https://github.com/pingcap/pd/CHANGELOG.md
+
+## [2.0.0-GA] - 2018-04-27
+### New Features
+* `tikv-ctl` now can recover mvcc data corruption.
```
Comment (lilin90) [inline CHANGELOG.md]: mvcc -> MVCC

File: CHANGELOG.md [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+* Support column families for raw key/value API.
+### Improves
+* Gc Raft logs cache eagerly to reduce memory usage.
+### Bug Fixes
+* fix a `tikv-ctl` bug maybe causes TiKV panic.
```
Comment (lilin90) [inline CHANGELOG.md]: -> Fix a `tikv-ctl` bug that might cause TiKV panic.

File: CHANGELOG.md [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+
+
+
+## [2.0.0-rc6] - 2018-04-19
+### New Features
+### Improves
+* Reduce lock contention in Worker.
+* Add metrics to the FuturePool.
+### Bug Fixes
+* Fix misused metrics in coprocessor.
```
Comment (lilin90) [inline CHANGELOG.md]: coprocessor -> Coprocessor

File: CHANGELOG.md [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+
+[tidb_change_log]: https://github.com/pingcap/tidb/CHANGELOG.md
+[pd_change_log]: https://github.com/pingcap/pd/CHANGELOG.md
+
+## [2.0.0-GA] - 2018-04-27
+### New Features
+* `tikv-ctl` now can recover mvcc data corruption.
+* Support column families for raw key/value API.
+### Improves
+* Gc Raft logs cache eagerly to reduce memory usage.
```
Comment (lilin90) [inline CHANGELOG.md]: Gc -> GC

File: CHANGELOG.md [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+### New Features
+### Improves
+* Reduce lock contention in Worker.
+* Add metrics to the FuturePool.
+### Bug Fixes
+* Fix misused metrics in coprocessor.
+
+## [2.0.0-rc.5] - 2018-04-17
+### New Features
+* Support compacting regions in `tikv-ctl`.
```
Comment (lilin90) [inline CHANGELOG.md]: regions -> Regions

File: CHANGELOG.md [hunk 10]
```diff
@@ -0,0 +1,132 @@
...
+* Add raw batch put/get/delete/scan API for TiKV service.
+* Add ImportKV service.
+* Support eval error in Coprocessor.
+* Support dynamic adjustment of RocksDB cache size by `tikv-ctl`.
+* Collect number of rows scanned for each range in Coprocessor.
+* Support treating overflow as warning in Coprocessor.
+* Support learner in raftstore.
+### Improves
+* Increase snap gc timeout.
```
Comment (lilin90) [inline CHANGELOG.md]: gc -> GC

File: CHANGELOG.md [hunk 11]
```diff
@@ -0,0 +1,132 @@
...
+* Speed up garbage cleaning when starting the server.
+* Update the size information about the corresponding Region according to compaction events.
+* Limit the size of scan lock to avoid request timeout.
+* Use DeleteRange to speed up Region deletion.
+
+## [2.0.0-rc.2] - 2018-03-15
+### New Features
+* Implement IngestSST API.
+* `tikv-ctl` now can send consistency-check requests to TiKV.
+* Support dumping stats of rocksdb and malloc in `tikv-ctl`.
```
Comment (lilin90) [inline CHANGELOG.md]: rocksdb -> RocksDB

File: CHANGELOG.md [hunk 12]
```diff
@@ -0,0 +1,132 @@
...
+* Fix the format compatibility issue of tikv-ctl proto.
+* Add timeout in Push metric.
+
+
+## [1.1.0-alpha] - 2018-01-19
+### New Features
+* Support Raft learner.
+* Support TLS.
+### Improves
+* Optimize Raft Snapshot and reduce the IO overhead.
```
Comment (lilin90) [inline CHANGELOG.md]: IO -> I/O

File: CHANGELOG.md [hunk 13]
```diff
@@ -0,0 +1,132 @@
...
+* Optimize the RocksDB configuration to improve performance.
+* Optimize count (*) and query performance of unique index in Coprocessor.
+* Solve the reconnection issue between PD and TiKV
+* Enhance the features of the data recovery tool `tikv-ctl`.
+* Support the Delete Range feature.
+* Support splitting according to table in regions.
```
Comment (lilin90) [inline CHANGELOG.md]: regions -> Regions

**Comment by siddontang** [comment]
PTAL @lilin90

File: CHANGELOG.md [hunk 14]
```diff
@@ -0,0 +1,132 @@
+# TiKV Change Log
+All notable changes to this project are documented in this file.
+See also [TiDB change log][tidb_change_log] and [PD change log][pd_change_log].
+
+[tidb_change_log]: https://github.com/pingcap/tidb/blob/change-log/CHANGELOG.md
+[pd_change_log]: https://github.com/pingcap/pd/blob/change-log/CHANGELOG.md
```
Comment (BusyJay) [inline CHANGELOG.md]: Wrong URL.

File: CHANGELOG.md [hunk 15]
```diff
@@ -0,0 +1,132 @@
...
+## [2.0.0-GA] - 2018-04-27
+### New Features
+* `tikv-ctl` now can recover MVCC data corruption.
+* Support column families for raw key/value API.
+### Improvements
+* Gc Raft logs cache eagerly to reduce memory usage.
+### Bug Fixes
+* Fix a `tikv-ctl` bug might cause TiKV panic.
+
+
```
Comment (BusyJay) [inline CHANGELOG.md]: Too many blank line.

File: CHANGELOG.md [hunk 16]
```diff
@@ -0,0 +1,132 @@
+# TiKV Change Log
+All notable changes to this project are documented in this file.
+See also [TiDB change log][tidb_change_log] and [PD change log][pd_change_log].
+
+[tidb_change_log]: https://github.com/pingcap/tidb/blob/change-log/CHANGELOG.md
+[pd_change_log]: https://github.com/pingcap/pd/blob/change-log/CHANGELOG.md
+
+## [2.0.0-GA] - 2018-04-27
```
Comment (BusyJay) [inline CHANGELOG.md]: Generally, GA will be omitted.

File: CHANGELOG.md (hunk 14 above)
Comment (lilin90) [inline CHANGELOG.md]: Still incorrect links?

File: CHANGELOG.md [hunk 17]
```diff
@@ -0,0 +1,132 @@
...
+### New Features
+* `tikv-ctl` now can recover MVCC data corruption.
+* Support column families for raw key/value API.
+### Improvements
+* Gc Raft logs cache eagerly to reduce memory usage.
```
Comment (lilin90) [inline CHANGELOG.md]: Gc -> GC

File: CHANGELOG.md [hunk 18]
```diff
@@ -0,0 +1,130 @@
+# TiKV Change Log
+All notable changes to this project are documented in this file.
+See also [TiDB change log][tidb_change_log] and [PD change log][pd_change_log].
+
+[tidb_change_log]: https://github.com/pingcap/tidb/blob/master/CHANGELOG.md
+[pd_change_log]: https://github.com/pingcap/pd/blob/master/CHANGELOG.md
+
+## [2.0.0] - 2018-04-27
+### New Features
+* `tikv-ctl` now can recover MVCC data corruption.
```
Comment (lilin90) [inline CHANGELOG.md]: I think the period in all the bulleted list items is unnecessary, to keep it consistent with other repos.

**Comment by siddontang** [comment]
PTAL @lilin90 @BusyJay

File: CHANGELOG.md [hunk 19]
```diff
@@ -0,0 +1,129 @@
+# TiKV Change Log
+All notable changes to this project are documented in this file.
+See also [TiDB change log][tidb_change_log] and [PD change log][pd_change_log].
+
+[tidb_change_log]: https://github.com/pingcap/tidb/blob/master/CHANGELOG.md
+[pd_change_log]: https://github.com/pingcap/pd/blob/master/CHANGELOG.md
+
+## [2.0.0] - 2018-04-27
```
Comment (BusyJay) [inline CHANGELOG.md]: Only add this section when 2.0.0 is about to be released.

**Comment by BusyJay** [comment]
LGTM

**Comment by BusyJay** [comment]
ping

File: CHANGELOG.md [hunk 20]
```diff
@@ -0,0 +1,120 @@
...
+* Configure the Raft snapshot max size to avoid the retained files taking up too much space
+* Support more recovery operations in tikv-ctl
+* Optimize the ordered flow aggregation operation
+
+## [1.0.8] - 2018-02-11
+### Improvements
+* Use DeleteFilesInRanges to clear stale data and improve the TiKV starting speed
+* Sync the metadata of the received Snapshot compulsorily to ensure its safety
+### Bug Fixes
+* Using Decimal in Coprocessor sum
```
Comment (lilin90) [inline CHANGELOG.md]: Using -> Use

**Pull request merged**